Model/Indexer/Pages.php

```diff
@@ -55,6 +60,10 @@ public function executeRow($slug)
 
     private function addPagesForSync($slugs)
     {
+
+        if ($this->storeAutocompleteConfig->hasToSyncPages()) {
```

Comment:
```suggestion
        if (!$this->storeAutocompleteConfig->hasToSyncPages()) {
```

We need to return from this function is it false.